source/common/http/http2/codec_impl.cc

```diff
@@ -381,10 +381,15 @@ ssize_t ConnectionImpl::StreamImpl::onDataSourceRead(uint64_t length, uint32_t*
     if (local_end_stream_ && pending_send_data_.length() <= length) {
       *data_flags |= NGHTTP2_DATA_FLAG_EOF;
       if (pending_trailers_to_encode_) {
-        // We need to tell the library to not set end stream so that we can emit the trailers.
-        *data_flags |= NGHTTP2_DATA_FLAG_NO_END_STREAM;
-        submitTrailers(*pending_trailers_to_encode_);
-        pending_trailers_to_encode_.reset();
+        const bool skip_encoding_empty_trailers =
```

Comment:
quick question for my own understanding: if we do skip encoding empty trailers, is it ok that they are not `reset()`? 
Just want to make sure it's not possible to trigger https://github.com/envoyproxy/envoy/blob/34fc31a98e6ecf2c8cd90efa8eeb6c56e02958d0/source/common/http/http2/codec_impl.cc#L230

Reply:
Good question. I’ll do some digging on the possibilities. However I think @alyssawilk could shed some light here.

Reply:
Yeah I we should add test on that (probably setting that null when we skip helps).

Reply:
I think you have pointed a valid concern. I pushed a change to make sure we do reset. Thanks, @asraa!